buildSrc/src/integTest/groovy/org/elasticsearch/gradle/YamlRestCompatTestPluginFuncTest.groovy

```diff
@@ -0,0 +1,114 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleFuncTest
+import org.gradle.testkit.runner.TaskOutcome
+import spock.lang.IgnoreIf
+
+class YamlRestCompatTestPluginFuncTest extends AbstractGradleFuncTest {
+
+    def "yamlRestCompatTest does nothing when there are no tests"() {
+        given:
+
+        addSubProject(":distribution:bwc:minor") <<  """
+        configurations { checkout }
+        artifacts {
+            checkout(new File(projectDir, "checkoutDir"))
+        }
+        """
+
+        buildFile << """
+        plugins {
+          id 'elasticsearch.yaml-rest-compat-test'
+        }
+        """
+
+        when:
+        def result = gradleRunner("yamlRestCompatTest").build()
+
+        then:
+        result.task(':yamlRestCompatTest').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyRestApiCompatSpecsTask').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyRestApiCompatTestTask').outcome == TaskOutcome.NO_SOURCE
+    }
+
+    def "yamlRestCompatTest executes and copies api and tests from :bwc:minor"() {
+        given:
+        String api = "wrongversion.json"
+        String test = "wrongversion.yml"
+        setupInternalRestResources(api, test)
+
+        addSubProject(":distribution:bwc:minor") <<  """
+        configurations { checkout }
+        artifacts {
+            checkout(new File(projectDir, "checkoutDir"))
+        }
+        """
+
+        //add the compatible test and api files
+        file("distribution/bwc/minor/checkoutDir/src/yamlRestTest/resources/rest-api-spec/test/foo/10_basic.yml") << ""
+        file("distribution/bwc/minor/checkoutDir/rest-api-spec/src/main/resources/rest-api-spec/api/foo.json") << ""
+
+        buildFile << """
+          apply plugin: 'elasticsearch.yaml-rest-compat-test'
+          import org.elasticsearch.gradle.testclusters.TestDistribution;
+          testClusters {
+            yamlRestCompatTest.setTestDistribution(TestDistribution.INTEG_TEST)
+          }
+        """
+
+        when:
+        def result = gradleRunner("yamlRestCompatTest").build()
+
+        then:
+        result.task(':yamlRestCompatTest').outcome == TaskOutcome.NO_SOURCE //no Java classes in source set
+        result.task(':copyRestApiCompatSpecsTask').outcome == TaskOutcome.SUCCESS
+        result.task(':copyRestApiCompatTestTask').outcome == TaskOutcome.SUCCESS
+
+        File resourceDir = new File(testProjectDir.root, "build/resources/yamlRestCompatTest/rest-api-spec")
+        // ensure the compatible resources from bwc:minor are in the resource dir
+        assert new File(resourceDir, "test/foo/10_basic.yml").exists()
```

Comment:
I don't believe we need `assert` here. Any boolean expression in a `then:` block is an implicit assertion.